### PR TITLE
Doc updates 032223

### PR DESCRIPTION
**Pull request by ronanstokes-db**
## Proposed changes

Cleanup of some key docs and addition of new content

## Types of changes

What types of changes does your code introduce to dbldatagen?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Change to tutorials, tests or examples
- [ ] Non code change (readme, images or other non-code assets)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. 
If you're unsure about any of them, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Submission does not reduce code coverage numbers
- [x] Submission does not increase alerts or messages from prospector / lint

## Further comments

Doc changes only

### CODE REVIEW

File: dbldatagen/column_generation_spec.py [hunk 1]
```diff
@@ -56,7 +56,8 @@ class ColumnGenerationSpec(object):
     :param distribution: Instance of distribution, that will control the distribution of the generated values
     :param baseColumn: String or list of strings representing columns used as basis for generating the column data
     :param randomSeed: random seed value used to generate the random value, if column data is random
-    :param randomSeedMethod: method for computing random values from the random seed
+    :param randomSeedMethod: method for computing random values from the random seed. It may take on the
+           values `fixed`, `hash_fieldname` or
```
Comment (howardwu-db) [inline dbldatagen/column_generation_spec.py:60]: Or what? Is this purposefully trailing here?

File: docs/source/options_and_features.rst [hunk 2]
```diff
@@ -12,6 +12,18 @@ Options for column specification
...
 
+Data generation for each column begins with generating a base expression for the column, which will be a random value
+if `random` is True, or some transformation of the `baseColumns` if not.
+
+The options are then applied - each option successively modifying the generated value. Where possible, the effects of
+an options are applied over the effects of other options so the effect is accumulative.
+
+Finally type conversion is applied.
+
+So options such as `minValue`, `maxValue` and `step` can be applied to strings also if their underlying generation is from
```
Comment (howardwu-db) [inline docs/source/options_and_features.rst:23]: Feels like a sentence fragment here

File: docs/source/options_and_features.rst [hunk 3]
```diff
@@ -25,16 +37,41 @@ step              Step to use for range of generated value.
...
+
+                  If `randomSeed` has a value of -1, then the random value will be a random number from the uniform
+                  distribution and the data generated will not be the same from run to run
+
+distribution      Controls the statistical distribution of random values when the column is generated
+                  randomly.
+                  Accepts the values "normal", or a Distribution object instance.
+
+baseColumn        Either the string name of the base column, or a list of columns to use to control
+                  data generation.
```
Comment (howardwu-db) [inline docs/source/options_and_features.rst:74]: Delete period here for consistency

File: docs/source/options_and_features.rst [hunk 4]
```diff
@@ -25,16 +37,41 @@ step              Step to use for range of generated value.
...
+                  the random seed for each column a hash value based on the field name. This is the default unless
+                  these settings are overridden
+
+
+                  If `randomSeed` has a value of -1, then the random value will be a random number from the uniform
+                  distribution and the data generated will not be the same from run to run
+
+distribution      Controls the statistical distribution of random values when the column is generated
+                  randomly.
+                  Accepts the values "normal", or a Distribution object instance.
```
Comment (howardwu-db) [inline docs/source/options_and_features.rst:71]: Delete period here for consistency

File: docs/source/options_and_features.rst [hunk 5]
```diff
@@ -25,16 +37,41 @@ step              Step to use for range of generated value.
...
+
+baseColumn        Either the string name of the base column, or a list of columns to use to control
+                  data generation.
 
 values            List of discrete values for the column.
```
Comment (howardwu-db) [inline docs/source/options_and_features.rst:76]: Delete period here for consistency

**Review by howardwu-db — CHANGES_REQUESTED** [review]

File: dbldatagen/column_generation_spec.py (hunk 1 above)
Comment (ronanstokes-db) [reply dbldatagen/column_generation_spec.py:60]: fixed

File: docs/source/options_and_features.rst (hunk 2 above)
Comment (ronanstokes-db) [reply docs/source/options_and_features.rst:23]: Fixed

**Review by howardwu-db — APPROVED** [review]

**Pull request merged**